containers/orchestration/app/utils.py

```diff
@@ -38,9 +48,32 @@ def load_processing_config(config_name: str) -> dict:
                 f"A config with the name '{config_name}' could not be found."
             )
 
+    # Replace placeholders with environment variable values, if present
+    replace_env_var_placeholders(processing_config)
+
     return processing_config
 
 
+def replace_env_var_placeholders(config: dict):
+    """
+    Check for environment variable placeholders in the configuration
+      and replace them if found.
+    :param config: Loaded config json file that needs to be checked for replace vars.
+    """
+    has_placeholders = False
+    # TODO: Currently, we are only replacing URLs, but may need to be
+    # more generalizable in the future
+    for settings in config.get("configurations", {}).values():
+        if "url" in settings and isinstance(settings["url"], str):
+            has_placeholders = True
+            break
+
+    if has_placeholders:
+        for settings in config.get("configurations", {}).values():
+            if "url" in settings and isinstance(settings["url"], str):
+                settings["url"] = os.path.expandvars(settings["url"])
```

Comment:
Having the same for loop in here twice `for settings in config.get("configurations", {}).values()` and checking this condition twice `if "url" in settings and isinstance(settings["url"], str)` seems like a code smell to me. I understand the theory of the first block to see if we can just skip this entire process, but I think we are over optmizing a bit here. So far at least one step will require `url` and so far all of these values need to be interpolated based on environment variables. I also don't think we need to ensure that the value of `url` is a string. This in theory should always be the case and if we are concerned about ensuring this we should handle it during a step where we validate config files before allowing the service to accept them.

What do you think about something like this?

```suggestion
        for settings in config.get("configurations", {}).values():
            if "url" in settings:
                settings["url"] = os.path.expandvars(settings["url"])
```

Reply:
makes sense to me. I'm definitely guilty of over-optimizing at times. Fixed in latest.